integration_tests/onedut_oneotg_tests/mpls_over_udp_scale/util/util.go

```diff
@@ -0,0 +1,353 @@
+}
+
+// MPLSLabelToPacketBytes converts an MPLS label (as uint32) into its 4-byte wire representation.
+// Assumes default EXP=0, BoS=1, TTL=255 for the generated bytes.
+func MPLSLabelToPacketBytes(label uint32) []byte {
+	buf := make([]byte, 4)
+	// Label (20 bits) << 12 | EXP (3 bits) << 9 | BoS (1 bit) << 8 | TTL (8 bits)
+	// Example: Label 100, EXP 0, BoS 1, TTL 255
+	// (100 << 12) | (0 << 9) | (1 << 8) | 255 = 0x000641FF
+	headerValue := (label << 12) | (1 << 8) | 255 // Set BoS=1, TTL=255
+	binary.BigEndian.PutUint32(buf, headerValue)
+	return buf
+}
+
+// populateNextHops generates NextHop entries and returns the updated slice.
+func populateNextHops(cfg *ScaleProfileConfig) ([]fluent.GRIBIEntry, error) {
+	totalNextHops := cfg.NumNexthopPerNHG * cfg.NumNexthopGroup
+	entries := []fluent.GRIBIEntry{}
+	if totalNextHops <= 0 {
+		return entries, nil
+	}
+
+	// Generate Destination IPs for Encap Header
+	dstIPs := []string{}
+	if cfg.NumDstIP > 0 {
+		startDstIP, err := netip.ParseAddr(cfg.DstIPStart)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse DstIPStart %q: %w", cfg.DstIPStart, err)
+		}
+		currentDstIP := startDstIP
+		for i := 0; i < cfg.NumDstIP; i++ {
+			if !currentDstIP.IsValid() {
+				return nil, fmt.Errorf("ran out of valid destination IP addresses starting from %s after %d IPs", cfg.DstIPStart, i)
+			}
+			dstIPs = append(dstIPs, currentDstIP.String())
+			currentDstIP = currentDstIP.Next()
+		}
+	} else {
+		return nil, errors.New("NumDstIP must be positive to generate destination IPs")
+	}
+
+	for i := 1; i <= totalNextHops; i++ {
+		mplsLabel := cfg.MPLSLabelStart
+		if !cfg.UseSameMPLSLabel {
+			mplsLabel = cfg.MPLSLabelStart + uint64(i-1)
+		}
+
+		nhEntry := fluent.NextHopEntry().
+			WithNetworkInstance(cfg.NetworkInstanceName).
+			WithIndex(uint64(i)).
+			WithIPAddress(cfg.EgressATEIPv6).
+			AddEncapHeader(
+				fluent.MPLSEncapHeader().WithLabels(mplsLabel),
+				// TODO: fluent currently doesn't support UDPV4 encap header builder. Support fluent.UDPV4EncapHeader to allow ipv4 support here.
+				fluent.UDPV6EncapHeader().
+					WithDstUDPPort(cfg.UDPDstPort).
+					WithSrcUDPPort(cfg.UDPSrcPort).
+					WithSrcIP(cfg.SrcIP).
+					WithDstIP(dstIPs[(i-1)%cfg.NumDstIP]).
+					WithDSCP(cfg.DSCP).
+					WithIPTTL(cfg.IPTTL),
+			)
+
+		entries = append(entries, nhEntry)
+	}
+	return entries, nil
+}
+
+// combinationKey generates a unique, sorted string key for a slice of NH indices.
+func combinationKey(indices []uint64) string {
+	sort.Slice(indices, func(i, j int) bool { return indices[i] < indices[j] })
+
+	var sb strings.Builder
+	for i, idx := range indices {
+		sb.WriteString(strconv.FormatUint(idx, 10))
+		if i < len(indices)-1 {
+			sb.WriteString(",")
+		}
+	}
+	return sb.String()
+}
+
+// populateNextHopGroups generates NextHopGroup entries, ensuring unique combinations of NHs per NHG.
+func populateNextHopGroups(cfg *ScaleProfileConfig) ([]fluent.GRIBIEntry, error) {
```

Comment:
nit: func (cfg * ScaleProfileConfig) populateNextHopGroups() for this and others